o1vm/src/interpreters/riscv32im/interpreter.rs

```diff
@@ -1752,7 +1752,21 @@ pub fn interpret_itype<Env: InterpreterEnv>(env: &mut Env, instr: IInstruction)
             env.set_next_instruction_pointer(next_instruction_pointer + Env::constant(4u32));
         }
         IInstruction::ShiftLeftLogicalImmediate => {
-            unimplemented!("ShiftLeftLogicalImmediate")
+            // slli: x[rd] = x[rs1] << shamt
+            let local_rs1 = env.read_register(&rs1);
+            let shamt = {
```

Comment:
Can you add some FIXME reg. [this comment](https://github.com/o1-labs/proof-systems/pull/2836#discussion_r1866012015) please? Can be done later.

Reply:
Note that a test is ready [in master](https://github.com/o1-labs/proof-systems/blob/master/o1vm/tests/test_riscv_elf.rs#L81) reg. sll, see [assembly code](https://github.com/o1-labs/proof-systems/blob/master/o1vm/resources/programs/riscv32im/src/sll.S). Only [lui](https://github.com/o1-labs/proof-systems/pull/2852) should be implemented.